scipy/signal/windows/windows.py

```diff
@@ -1652,6 +1652,63 @@ def exponential(M, center=None, tau=1., sym=True):
 
     return _truncate(w, needs_trunc)
 
+def taylor(N, nbar=4, level=-30):
+    """
+    Return the Taylor window.
+
+    The Taylor window allows for a selectable sidelobe suppression with a 
+    minimum broadening. This window is commonly used in radar processing [1].
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an
+        empty array is returned.
+    nbar : int
+        Number of nearly constant level sidelobes adjacent to the mainlobe
+    level : float
+        Desired peak sidelobe level in decibels (db) relative to the mainlobe
+
+    Returns
+    -------
+    out : array
+        The window, with the center value normalized to one (the value
+        one appears only if the number of samples is odd).
+
+    See Also
+    --------
+    kaiser, bartlett, blackman, hamming, hanning
+
+    References
+    -----
+    .. [1] W. Carrara, R. Goodman, and R. Majewski "Spotlight Synthetic 
+               Aperture Radar: Signal Processing Algorithms" Pages 512-513,
+               July 1995.
+    """
+    B = 10**(-level / 20)
+    A = np.log(B + np.sqrt(B**2 - 1)) / np.pi
+    s2 = nbar**2 / (A**2 + (nbar - 0.5)**2)
+    ma = np.arange(1,nbar)
+    
+    def calc_Fm(m):
+        numer = (-1)**(m+1) * np.prod(1 - m**2/s2/(A**2 + (ma - 0.5)**2))
+        denom = 2 * np.prod([1 - m**2/j**2 for j in ma if j != m])
+        return numer/denom
+
+    calc_Fm_vec = np.vectorize(calc_Fm)
+    Fm = calc_Fm_vec(ma)
```

Comment:
Since the `calc_Fm` code is only effectively used once, instead of making a function and doing `np.vectorize`, it would be simpler to make a loop. Something along these lines:
```
Fm = np.empty(nbar-1)
for mi, m in enumerate(ma):
    numer = (-1)**(m+1) * np.prod(1 - m**2/s2/(A**2 + (ma - 0.5)**2))
    denom = 2 * np.prod([1 - m**2/j**2 for j in ma if j != m])
    Fm[mi] = numer / denom
```
And also note that you can start taking steps toward vectorizing if you want, e.g. with:
```
Fm = np.empty(nbar-1)
signs = np.where(ma % 2, 1, -1)
m2 = ma*ma
for mi, m in enumerate(ma):
    numer = signs[mi] * np.prod(1 - m2[mi]/s2/(A**2 + (ma - 0.5)**2))
    denom = 2 * np.prod([1 - m2[mi]/m_ for m_ in m2 if m_ != m2[mi]])
    Fm[mi] = numer / denom
```
And then get rid of iteration in the `denom`:
```
    Fm = np.empty(nbar-1)
    signs = np.where(ma % 2, 1, -1)
    m2 = ma*ma
    for mi, m in enumerate(ma):
        numer = signs[mi] * np.prod(1 - m2[mi]/s2/(A**2 + (ma - 0.5)**2))
        denom = 2 * np.prod(1 - m2[mi]/m2[:mi]) * np.prod(1 - m2[mi]/m2[mi+1:])
        Fm[mi] = numer / denom
```
There is probably a way to fully vectorize this (get rid of the loop), but in the meantime this should at least give some speedup.

Reply:
Implemented your final version here.